app/src/main/java/org/alphatilesapps/alphatiles/Start.java

```diff
@@ -1548,6 +1608,235 @@ public ArrayList<String> parseWordIntoTiles(String parseMe) {
             return parsedWordArrayTemp;
         }
 
+        /*
+        LM: parseWordIntoTiles does not currently take into consideration symbols of type X because we don't have
+        data to show how these tiles pattern in Thai script language packs yet. 03-2023
+         */
+        public ArrayList<String> parseWordIntoTiles(String stringToParse, String wordInLOP) {
+            ArrayList<String> parsedWordArrayPreliminary = parseWordIntoTilesPreliminary(stringToParse);
+            ArrayList<String> parsedWordArrayPreliminaryReference = parseWordIntoTilesPreliminary(wordInLOP);
+
+            if (!(scriptType.equals("Thai") || scriptType.equals("Lao"))) {
+                return parsedWordArrayPreliminary;
+            }
+
+            ArrayList<String> parsedWordArray = new ArrayList<String>();
+
+            int consonantScanIndex = 0;
+            int currentConsonantIndex = 0;
+            int previousConsonantIndex = -1;
+            int nextConsonantIndex = parsedWordArrayPreliminary.size();
+            boolean foundNextConsonant = false;
+
+            while (consonantScanIndex < parsedWordArrayPreliminary.size()) {
+
+                String currentConsonant = "";
+
+                foundNextConsonant = false;
+                Tile currentTile;
+                String currentTileString;
+                String currentTileType;
+                // Scan for the next unchecked consonant tile
+                while (!foundNextConsonant && consonantScanIndex < parsedWordArrayPreliminary.size()) {
+                    currentTileString = parsedWordArrayPreliminary.get(consonantScanIndex);
+                    currentTile = tileHashMap.get(currentTileString);
+                    if (MULTIFUNCTIONS.contains(currentTileString)) { // Discern the type of the current tile
+                        int indexInPreliminaryArray = returnInstanceIndexInPreliminaryParsedWordArray(currentTileString, consonantScanIndex, parsedWordArrayPreliminary, wordInLOP);
```

Comment:
I will look into this, as well as your previous comment. I agree that the current structure is cumbersome. Changing to using Tile lists rather than String lists will require refactors in all of the game classes for accessing the Strings as properties of Tiles instead of directly, so I will look at the tradeoffs.